pyproject.toml

```diff
@@ -122,3 +122,12 @@ command_line = '-m pytest'
 # TODO: Cannot report relative to project path because we cannot do editable installs yet
 #source = ['python/spglib']
 source = ['spglib']
+
+[tool.ruff]
+line-length = 88
+exclude = [
+    "database",
+]
+
+[tool.mypy]
+no_strict_optional = true
```

Comment:
mypy is not so smart to pass
```
d: dict | None
assert(d is not None)
d.keys()
```

Reply:
It did on various projects :thinking:. I don't see anything special in the configure either:
```toml
[tool.mypy]
python_version = "3.11"
warn_unused_configs = true
show_error_codes = true
strict = true
disallow_untyped_defs = false
disallow_incomplete_defs = false
```

Reply:
Do you recommend these configs that you write?

Reply:
The strict mode is too strict now. But, I'll add other flags because they looks useful.

Reply:
I would add `strict` as a comment with a todo/open PR

Reply:
Ok, I will leave todo as comment.